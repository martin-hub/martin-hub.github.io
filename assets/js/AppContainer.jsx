import React from 'react';
import styled from 'styled-components';
import {media} from './media.jsx';

/**
 * This is the root component of the app, styling the page background & vertical padding responsively
 */
const AppContainer = styled.div`
  background-color: #0D0E0F;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  max-width: 100%;

  ${media.xxlarge`
    padding: 90px 0;
  `}

  ${media.xlarge`
    padding: 70px 0;
  `}

  ${media.large`
    padding: 35px 0;
  `}

  ${media.medium`
    padding: 0;
  `}

  ${media.small`
    padding: 0;
  `}

  &,
  * {
    box-sizing: border-box;
    font-family: sans-serif;
  }
`;

export default AppContainer;
