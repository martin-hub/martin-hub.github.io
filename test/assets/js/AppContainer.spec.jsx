import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import AppContainer from 'AppContainer.jsx';

describe("AppContainer", () => {
  test('renders correctly', () => {
    expect(renderer.create(<AppContainer />).toJSON()).toMatchSnapshot();
  });
});