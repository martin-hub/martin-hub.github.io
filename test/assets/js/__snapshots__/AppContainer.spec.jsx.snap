// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer renders correctly 1`] = `
.c0 {
  background-color: #0D0E0F;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  max-width: 100%;
}

.c0,
.c0 * {
  box-sizing: border-box;
  font-family: sans-serif;
}

@media (min-width:120.0625em) {
  .c0 {
    padding: 90px 0;
  }
}

@media (max-width:120em) {
  .c0 {
    padding: 70px 0;
  }
}

@media (max-width:90em) {
  .c0 {
    padding: 35px 0;
  }
}

@media (max-width:75.625em) {
  .c0 {
    padding: 0;
  }
}

@media (max-width:47.9375em) {
  .c0 {
    padding: 0;
  }
}

<div
  className="c0"
/>
`;
